cpp/include/raft/cluster/detail/kmeans_balanced.cuh

```diff
@@ -128,7 +129,7 @@ inline std::enable_if_t<std::is_floating_point_v<MathT>> predict_core(
 
       // todo(lsugy): use KVP + iterator in caller.
       // Copy keys to output labels
-      thrust::transform(resource::get_thrust_policy(handle),
+      thrust::transform(resource::get_thrust_nosync_policy(handle),
                         minClusterAndDistance.data_handle(),
                         minClusterAndDistance.data_handle() + n_rows,
                         labels,
```

Comment:
This could be done with
```
raft::linalg::map(handle, minClusterAndDistance, raft::compose_op<raft::cast_op<LabelT>, raft::key_op>{}, minClusterAndDistance);
```